tests/integration/test_api_gateway.py

```diff
@@ -127,3 +148,38 @@ def forward_request(self, **kwargs):
 
     # clean up
     proxy.stop()
+
+
+def test_api_gateway_lambda_proxy_integration():
+    test_port = 12123
+    backend_url = 'http://localhost:%s%s' % (test_port, API_PATH_LAMBDA_PROXY_BACKEND)
+
+    # create target HTTP backend
+    class TestListener(ProxyListener):
```

Comment:
From my understanding, this PR is supposed to test the Lambda proxy integration. So we basically need to construct the following request flow:

Client request -> API Gateway -> Lambda function

So, as part of the test we need to create:
* the API Gateway resources
* a simple Lambda function

The test should then assert that the client request is successfully forwarded to the Lambda function. The question is: how we can make that assertion, because we need some kind of "shared state" between the test execution and the Lambda function. There are several possible options, but one of the simplest I can think of is to simply create an S3 bucket with a distinct name from within the Lambda function, and then assert in the test script (`test_api_gateway.py`) that the bucket has actually been created.

If we go with the option of creating the S3 bucket, then it appears that there is no need for a HTTP backend service in this test. Therefore, we can probably remove the `TestListener` class from here, am I right?